lib/file_store/adapters/memory.ex

```diff
@@ -130,6 +130,18 @@ defmodule FileStore.Adapters.Memory do
       end)
     end
 
+    def copy(store, src, dest) do
+      with {:ok, content} <- read(store, src) do
+        write(store, dest, content)
+      end
+    end
+
+    def move(store, src, dest) do
+      with :ok <- copy(store, src, dest) do
+        delete(store, src)
+      end
+    end
```

Comment:
It looks like this implementation will work just fine; however, with just a little bit more code, these agent operations can be made more efficient:

```suggestion
    def copy(store, src, dest) do
      Agent.get_and_update(store.name, fn state ->
        case Map.fetch(state, src) do
          {:ok, value} ->
            {:ok, Map.put(state, dest, value)}

          :error ->
            {{:error, :enoent}, state}
        end
      end)
    end

    def move(store, src, dest) do
      Agent.get_and_update(store.name, fn state ->
        case Map.fetch(state, src) do
          {:ok, value} ->
            {:ok, state |> Map.delete(src) |> Map.put(dest, value)}

          :error ->
            {{:error, :enoent}, state}
        end
      end)
    end
```

Reply:
Incorporated this change.